src/codegen/codegen_c_visitor.cpp

```diff
@@ -1606,12 +1606,21 @@ void CodegenCVisitor::print_table_replacement_function(const ast::Block& node) {
     print_function_declaration(node, name);
     printer->start_block();
     {
-        printer->add_line("if ( {} == 0) {}"_format(use_table_var, "{"));
-        printer->add_line("    {}({}, arg_v);"_format(function_name, internal_method_arguments()));
+        const auto& params = node.get_parameters();
+        if (params.size() != 1) {
+            throw std::runtime_error(
+                "FUNCTION or PROCEDURE containing a TABLE stmt must have exactly one argument");  // Check it in parser?
+        }
```

Comment:
nitpick - we can remove this from here as your semantic visitor will do this now. (Or could be just assert here)